Doc/library/sqlite3.rst

```diff
@@ -47,84 +47,148 @@ This document includes four main sections:
+by executing an ``INSERT`` statement,
+once again by calling :meth:`~Cursor.execute` on ``cur``::
 
-At this point, our database only contains one row::
+   cur.execute("""
+       INSERT INTO Movie VALUES
+           ('Monty Python and the Holy Grail', 1975, 8.2),
+           ('And Now for Something Completely Different', 1971, 7.5)
+   """)
 
-   >>> res = cur.execute('SELECT count(rowid) FROM stocks')
-   >>> print(res.fetchone())
-   (1,)
+The ``INSERT`` statement implicitly opens a transaction,
+which needs to be committed before changes are saved in the database
+(see :ref:`sqlite3-controlling-transactions` for details).
+Call :meth:`~Connection.commit` on the connection object
+to commit the transaction::
 
-The result is a one-item :class:`tuple`:
-one row, with one column.
-Now, let us insert three more rows of data,
-using :meth:`~Cursor.executemany`::
+   con.commit()
 
-   >>> data = [
-   ...    ('2006-03-28', 'BUY', 'IBM', 1000, 45.0),
-   ...    ('2006-04-05', 'BUY', 'MSFT', 1000, 72.0),
-   ...    ('2006-04-06', 'SELL', 'IBM', 500, 53.0),
-   ... ]
-   >>> cur.executemany('INSERT INTO stocks VALUES(?, ?, ?, ?, ?)', data)
+Execute a query to verify that the data was inserted correctly.
+Use the now familiar :meth:`~Cursor.execute` method on ``con``,
+store the result in ``res``,
+and call :meth:`~Cursor.fetchall` on ``res`` to fetch all rows::
+
+   >>> res = cur.execute("SELECT Score FROM Movie")
+   >>> res.fetchall()
+   [(8.2,), (7.5,)]
+
+The result is a :class:`list` with two rows of data,
+each a one-item :class:`tuple`,
+the single item representing the column queried.
+
+Now, insert three more rows by calling
+:meth:`~Cursor.executemany` on the cursor object,
+once again committing the transaction by calling :meth:`~Connection.commit`
+on the connection object::
+
+   data = [
+       ("Monty Python Live at the Hollywood Bowl", 1982, 7.9),
+       ("Monty Python's The Meaning of Life", 1983, 7.5),
+       ("Monty Python's Life of Brian", 1979, 8.0),
+   ]
+   cur.executemany("INSERT INTO Movie VALUES(?, ?, ?)", data)
+   con.commit()
 
-Notice that we used ``?`` placeholders to bind *data* to the query.
+Notice that ``?`` placeholders are used to bind ``data`` to the query.
 Always use placeholders instead of :ref:`string formatting <tut-formatting>`
 to bind Python values to SQL statements,
 to avoid `SQL injection attacks`_.
-See the :ref:`placeholders how-to <sqlite3-placeholders>` for more details.
+See :ref:`sqlite3-placeholders` for more details.
 
-Then, retrieve the data by iterating over the result of a ``SELECT`` statement::
+Verify that data has been committed and written to disk by
+closing the connection and opening a new one,
+creating a new cursor,
+and then executing a ``SELECT`` query to read from the database::
 
-   >>> for row in cur.execute('SELECT * FROM stocks ORDER BY price'):
+   >>> con.close()
+   >>> con2 = sqlite3.connect("tutorial.db")
+   >>> cur2 = con2.cursor()
```

Comment:
This part seems misleading.  Before we didn't have to close and reopen the connection to verify that the table had been created, so why do we need to do it now?  Are changes not saved until you close the connection?  Or are you just trying to introduce the `connection.close` method and showing how to connect to an existing db?

If it's the latter, I wouldn't "hide" it in the middle of a verification, but I would move it at the end of the tutorial.  Once everything is done, close the db and mention that the data is permanently stored in the file, and that you can reconnect and retrieve it at any moment (possibly showing an example of another `SELECT`).

Reply:
The SELECT is leftovers from the earlier revision.

The point of this step is to verify that changes are written to disk, not that the table has been created (we already did that). I'll try to make that clearer.

Reply:
Introducing connection.close() should also be done. I'll add that as well.

Reply:
Rewritten, however I believe it came out a little bulky. Let me know what you think.

Reply:
Yeah, it is rather dense. I would just split it up into two chunks again, but with separate, appropriate descriptive prose for each part.

Reply:
Introducing the cursor iterator earlier will help; then we'd only need the users attention for the verification here.